mlserver/codecs/pandas.py

```diff
@@ -13,13 +13,17 @@
 
 def _to_series(input_or_output: InputOrOutput) -> pd.Series:
     payload = get_decoded_or_raw(input_or_output)
+
+    if input_or_output.datatype == "BYTES":
```

Comment:
Good catch! 

The `to_dtype` method infers, based on the shape, the byte size of each element in the array. It's probably best to leave it as you have now, to ensure that behaviour is the same as before, but I agree that we should review it at some point.